### ISSUE TITLE
[FEA] Nice to have `GeoSeries.loc`

### ISSUE THREAD

**Issue by thomcom** [rapidsai/cuspatial#592]
Depends on #585.

We need to support the more abstract slicing techniques, not just integer `ILoc`

```py
import geopandas
import cuspatial
df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))
gdf = cuspatial.from_geopandas(df)
gdf.loc[[1, 0]] # NotImplementedError
```

Optimally we can do:

```py
gdf.loc[['a', 'b']]
gdf.loc[[1, 0]]
```

etc

**Comment by thomcom** [comment]
Done
